src/main/java/io/github/deweyjose/graphqlcodegen/Validations.java

```diff
@@ -0,0 +1,61 @@
+package io.github.deweyjose.graphqlcodegen;
+
+import java.io.File;
+import java.util.*;
+
+public class Validations {
+
+  /**
+   * We sort the input files by their paths to ensure a consistent order for processing.
+   *
+   * We maintain two sets:
+   *   encounteredDirectories to keep track of encountered directories
+   *   encounteredPaths to keep track of encountered paths.
+   *
+   * We iterate through the sorted files and perform the following checks:
+   *   Directory overlap: throw an exception if duplicate directories are configured.
+   *   Duplicate files: throw an exception if duplicate files are configured.
+   *   Ancestry in already added directories:
+   *    if files exist in configured directories or subdirectories of configured directories
+   *    throw an exception
+   * @param files
+   */
+  public static void verifySchemaPaths(Collection<File> files) {
+    // Sort the input files to ensure consistent order
+    List<File> sortedFiles = new ArrayList<>(files);
+    sortedFiles.sort(Comparator.comparing(File::getPath));
+
+    Set<File> encounteredDirectories = new HashSet<>();
+    Set<String> encounteredPaths = new HashSet<>();
+
+    for (File file : sortedFiles) {
+      String path = file.getPath();
+
+      // Check for directory overlap
+      File parent = file.getParentFile();
+
+      // Check for ancestry in already added directories
+      while (parent != null) {
+        if (encounteredDirectories.contains(parent)) {
+          throw new IllegalArgumentException(
+            String.format("Overlap configured: %s contains %s", parent.getPath(), file.getPath())
```

Comment:
@mjfaga This one catches the case where I've configured /dir/foo and /dir/foo/a.graphqls. DGS will happily accept both, and will add all graphqls files under /dir/foo which will process a.graphqls twice causing Java compile errors later with generated code.

I think we want to fault here and let folks fix the config, either remove /dir/foo or /dir/foo/a.graphqls. It's unclear what they want here, the single file or all the files under foo.